mantidimaging/core/fitting/fitting_engine.py

```diff
@@ -22,17 +22,44 @@ def get_parameter_names(self) -> list[str]:
     def get_init_params_from_roi(self, region: FittingRegion) -> dict[str, float]:
         return self.model.get_init_params_from_roi(region)
 
-    def find_best_fit(self, xdata: np.ndarray, ydata: np.ndarray, initial_params: list[float]) -> dict[str, float]:
-        additional_params = self.model.prefitting(xdata, ydata, initial_params)
-        params_to_fit = initial_params[:len(initial_params) - len(additional_params)]
+    def find_best_fit(
+        self,
+        xdata: np.ndarray,
+        ydata: np.ndarray,
+        initial_params: list[float],
+        sigma: np.ndarray | None = None,
```

Comment:
I think we should hold off doing any weighted measures until we have an option to turn on a weighted least squares fit. (There are cases where there is not a good estimate of errors, so it should be an option).

So we should probably be careful with the labelling (like you have done with "SSE"). "RSS" (residual sum of squares) might be a bit more accurate, because we only know they are residuals.

RSS/DoF is probably useful, so that we have something that scales with the number of data points. With DoF as `ydata.size - len(all_params)`